lib/trip_dispatcher.rb

```diff
@@ -72,5 +100,35 @@ def inspect
     def check_id(id)
       raise ArgumentError, "ID cannot be blank or less than zero. (got #{id})" if id.nil? || id <= 0
     end
+
+    def request_trip(user_id)
+      new_user = find_passenger(user_id)
+      driver = @driver.select {|driver| driver.status == :AVAILABLE}
+
```

Comment:
I had trouble calling this method. You've got it in the `private` section of this class (see line 98 above), which means it's not accessible from outside.

Once I removed `private`, I found the logic was pretty close, but I ran into a couple of problems:

- On line 106, `@driver` should be `@drivers`
- `.select` returns an array, but you store it in a variable called `driver` (implying just one) and use it like it's an instance of `Driver`. `.find` might be what you're looking for here.
- The `Trip` class doesn't yet know how to handle a `nil` end time

The rest of the logic in this method looks great. My guess is that if you had been able to run this method you would have cruise through the rest of these problems. This would have been a good time to come to an instructor for help.